python-checks/src/main/java/org/sonar/python/checks/UseOfAnyAsTypeHintCheck.java

```diff
@@ -42,14 +52,52 @@ public void initialize(Context context) {
 
   private static void checkForAnyInTypeHint(SubscriptionContext ctx) {
     TypeAnnotation typeAnnotation = (TypeAnnotation) ctx.syntaxNode();
-    if (isTypeAny(typeAnnotation)) {
-      ctx.addIssue(typeAnnotation.expression(), MESSAGE);
-    }
+    Optional.of(typeAnnotation)
+      .filter(UseOfAnyAsTypeHintCheck::isTypeAny)
+      .filter(Predicate.not(UseOfAnyAsTypeHintCheck::isAnnotatingArgumentToChildMethod))
+      .filter(Predicate.not(UseOfAnyAsTypeHintCheck::isUnannotatedOverride))
```

Comment:
Feel free to remove the negation of the predicate here. For some reason I just really didn't want to have `isNot...` at the beginning of the method name.